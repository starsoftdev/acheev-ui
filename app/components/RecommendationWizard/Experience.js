// @flow
import * as React from 'react';

import Icon from 'components/Icon';
import NewIcon from 'images/sprite/new_experience.svg';
import FamiliarIcon from 'images/sprite/familiar_experience.svg';
import ConnoisseurIcon from 'images/sprite/connoisseur_experience.svg';

import Block from './Block';

const options = [
  {
    icon: NewIcon,
    label: 'New to cannabis',
    field: 'new',
  },
  {
    icon: FamiliarIcon,
    label: 'Familiar with cannabis',
    field: 'familiar',
  },
  {
    icon: ConnoisseurIcon,
    label: 'A cannabis connoisseur',
    field: 'connoisseur',
  },
];

type Props = {
  onSelect: (value: string) => void,
  // eslint-disable-next-line  react/no-unused-prop-types
  value: string | Array<string>,
};

type State = {
  value?: string,
};

export default class Experience extends React.Component<Props, State> {
  state = {
    value: undefined,
  };

  handleClick = (e: SyntheticEvent<HTMLInputElement>) => {
    const { name } = e.currentTarget.dataset;
    this.setState({ value: name });
    this.props.onSelect(name);
  };

  render() {
    return (
      <section className="recommendationWizard__body">
        <div className="container">
          <div className="row">
            <h1 className="recommendationWizard__title small-12 column">
              I am
            </h1>
            <div className="small-12 column">
              <div className="row">
                <div className="small-12 large-10 large-offset-1 column">
                  <div className="row">
                    {options.map(item => (
                      <div key={item.field} className="small-12 large-4 column">
                        <Block
                          label={item.label}
                          onClick={this.handleClick}
                          data-name={item.field}
                          selected={item.field === this.state.value}
                        >
                          <Icon
                            glyph={item.icon}
                            width={100}
                            height={35}
                            className="recommendationWizard__icon"
                          />
                        </Block>
                      </div>
                    ))}
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    );
  }
}
