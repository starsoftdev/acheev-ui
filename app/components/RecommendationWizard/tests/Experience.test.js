import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import Experience from 'components/RecommendationWizard/Experience';

describe('<Experience />', () => {
  it('should match snapshot', () => {
    const component = create(<Experience onSelect={() => {}} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should call onSelect', () => {
    const handleSelect = jest.fn();
    const component = shallow(<Experience onSelect={handleSelect} />);

    component
      .find('Block')
      .first()
      .simulate('click', { currentTarget: { dataset: { name: 'new' } } });

    expect(handleSelect).toHaveBeenCalledWith('new');
    expect(handleSelect).toHaveBeenCalledTimes(1);
  });
});
