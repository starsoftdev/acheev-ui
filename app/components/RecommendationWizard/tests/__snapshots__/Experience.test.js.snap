// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Experience /> should match snapshot 1`] = `
<section
  className="recommendationWizard__body"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <h1
        className="recommendationWizard__title small-12 column"
      >
        I am
      </h1>
      <div
        className="small-12 column"
      >
        <div
          className="row"
        >
          <div
            className="small-12 large-10 large-offset-1 column"
          >
            <div
              className="row"
            >
              <div
                className="small-12 large-4 column"
              >
                <div
                  className="recommendationWizard__block"
                  data-name="new"
                  onClick={[Function]}
                >
                  <svg
                    className="recommendationWizard__icon"
                    height={35}
                    width={100}
                  >
                    <use
                      xlinkHref="#undefined"
                    />
                  </svg>
                  <p
                    className="recommendationWizard__label"
                  >
                    New to cannabis
                  </p>
                </div>
              </div>
              <div
                className="small-12 large-4 column"
              >
                <div
                  className="recommendationWizard__block"
                  data-name="familiar"
                  onClick={[Function]}
                >
                  <svg
                    className="recommendationWizard__icon"
                    height={35}
                    width={100}
                  >
                    <use
                      xlinkHref="#undefined"
                    />
                  </svg>
                  <p
                    className="recommendationWizard__label"
                  >
                    Familiar with cannabis
                  </p>
                </div>
              </div>
              <div
                className="small-12 large-4 column"
              >
                <div
                  className="recommendationWizard__block"
                  data-name="connoisseur"
                  onClick={[Function]}
                >
                  <svg
                    className="recommendationWizard__icon"
                    height={35}
                    width={100}
                  >
                    <use
                      xlinkHref="#undefined"
                    />
                  </svg>
                  <p
                    className="recommendationWizard__label"
                  >
                    A cannabis connoisseur
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
